eng/packages/http-client-csharp/generator/Azure.Generator/test/Providers/ResourceProviderTests.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Generator.Providers;
+using Azure.Generator.Tests.Common;
+using Azure.Generator.Tests.TestHelpers;
+using Microsoft.TypeSpec.Generator.Primitives;
+using NUnit.Framework;
+using System.Linq;
+
+namespace Azure.Generator.Tests.Providers
+{
+    internal class ResourceProviderTests
+    {
+        [TestCase]
```

Comment:
Can we add or create follow up issues to test the various types of resources?  parent, grandparent, singleton, etc?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/48431